src/dmd/backend/machobj.d

```diff
@@ -2421,7 +2421,15 @@ void MachObj_lidata(int seg,targ_size_t offset,targ_size_t count)
 {
     //printf("MachObj_lidata(%d,%x,%d)\n",seg,offset,count);
     size_t idx = SegData[seg].SDshtidx;
-    if ((I64 ? SecHdrTab64[idx].flags : SecHdrTab[idx].flags) == S_ZEROFILL)
+
+    // printf("SecHdrTab64[idx] = %s/%s\n", SecHdrTab64[idx].segname.ptr, SecHdrTab64[idx].sectname.ptr);
+    // printf("SecHdrTab64[idx].flags = %u\n", SecHdrTab64[idx].flags);
+    // printf("SecHdrTab64[idx].offset = %u\n", SecHdrTab64[idx].offset);
+    // printf("SegData[seg].SDoffset = %zu\n", SegData[seg].SDoffset);
+    // fflush(stdout);
+
+    const flags = (I64 ? SecHdrTab64[idx].flags : SecHdrTab[idx].flags);
+    if (flags == S_ZEROFILL || flags == S_THREAD_LOCAL_ZEROFILL)
```

Comment:
Doesn't the error message imply the check should be `if(!(flags == S_ZEROFILL...))`?

Reply:
Don't think so. The `if` deals with zero-initialised sections and the section referenced in the error message is marked as `S_THREAD_LOCAL_ZEROFILL`